### PR TITLE
PLAT-135588: Update Heading

### PR DESCRIPTION
**Pull request by stanca-pop-lgp**
Enact-DCO-1.0-Signed-off-by: Stanca Pop stanca.pop@lgepartner.com

### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [x] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [x] Documentation was verified or is not changed
* [x] UI test was passed or is not needed
* [x] Screenshot test was verified or is not needed
* [ ] This is an API breaking change

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Updated Heading according to the new design requirements. Updated styling, added `showBackButton` prop, updated unit and screenshot tests. For now, the back button's icon will be rendered conditionally, depending on the skin, as Silicon is the only skin that supports `arrowleft` icon. For the other the 'arrowlargeleft' will be used.

### Links
[//]: # (Related issues, references)
PLAT-135588

### Comments

### CODE REVIEW

**Review by daniel-stoian-lgp — CHANGES_REQUESTED** [review]
Check margin-left, margin-right from UX guide
UX guide says silicon Heading should have no padding
font-size should be 42px for all sizes in silicon, according to the UX guide

**Comment by daniel-stoian-lgp** [comment]
Since Heading max-witdth is modified in Silicon and Heading is used in the Storybook app in all samplers, maybe we'll need to adjust the width of the <Heading> in ThemeEnvironment, but I would like to know other reviewer's opinion.

File: Heading/Heading.js [hunk 1]
```diff
@@ -15,14 +15,18 @@
  * @exports HeadingDecorator
  */
 
+
```
Comment (daniel-stoian-lgp) [inline Heading/Heading.js:18]: remove empty line

File: Heading/Heading.js [hunk 2]
```diff
@@ -55,6 +59,15 @@ const HeadingBase = kind({
 
 		css: PropTypes.object,
 
+		/**
+		 * Shows the back button.
+		 *
+		 * @type {Boolean}
+		 * @default false
```
Comment (daniel-stoian-lgp) [inline Heading/Heading.js:66]: remove line

**Comment by stanca-pop-lgp** [comment]
> font-size should be 42px for all sizes in silicon, according to the UX guide

I think "large", "medium" and "small" from UX guide are more likely values for the "spacing" prop, rather than "size".

File: styles/variables-silicon.less [hunk 3]
```diff
@@ -121,7 +121,11 @@
 
 // Heading
 // ---------------------------------------
-@agate-heading-padding: 3px 0;
+@agate-heading-height: 60px;
+@agate-heading-margin: 0;
+@agate-heading-padding: 0 (2.5 * @agate-spotlight-outset) 0 @agate-spotlight-outset;
```
Comment (daniel-stoian-lgp) [inline styles/variables-silicon.less:126]: I would use static values here : 0 30px 0 9px;
With 9px we are closer to the UX requirement of 10px

**Review by daniel-stoian-lgp — APPROVED** [review]

**Review by jeonghee27 — APPROVED** [review]
LGTM

File: Heading/Heading.module.less [hunk 4]
```diff
@@ -15,6 +23,17 @@
 		text-transform: @agate-heading-text-transform;
 		font-variant-ligatures: @agate-heading-font-ligatures;
 
+		.backButton {
+			.bg {
+				border: none;
+				background-color: transparent;
+			}
```
Comment (MikyungKim) [inline Heading/Heading.module.less:30]: We have `backgroundOpacity="transparent"` prop instead of this.

File: Heading/Heading.module.less [hunk 5]
```diff
@@ -15,6 +23,17 @@
...
+
+			.client {
+				color: @agate-heading-text-color
+			}
```
Comment (MikyungKim) [inline Heading/Heading.module.less:34]: What is this color for?

File: styles/variables-silicon.less [hunk 6]
```diff
@@ -141,7 +141,11 @@
 
 // Heading
 // ---------------------------------------
-@agate-heading-padding: 3px 0;
+@agate-heading-height: 60px;
+@agate-heading-margin: 0;
+@agate-heading-padding: 0 30px 0 9px;
+@agate-heading-line-height: @agate-heading-height;
+@agate-heading-width: 660px;
```
Comment (MikyungKim) [inline styles/variables-silicon.less:148]: I don't think this is necessary. It looks natural when the heading has no width and the container has width.
Please remove it.

File: Heading/Heading.module.less [hunk 7]
```diff
@@ -5,7 +5,15 @@
 @import "../styles/skin.less";
 
 .heading {
+	.backButton {
+		align-items: center;
+		display: flex;
+		justify-content: center;
```
Comment (MikyungKim) [inline Heading/Heading.module.less:11]: `justify-content` This property isn't needed. Is this necessary?

File: Heading/Heading.module.less (hunk 4 above)
Comment (stanca-pop-lgp) [reply Heading/Heading.module.less:30]: Yes, you are right. Thank you!

File: Heading/Heading.module.less (hunk 5 above)
Comment (stanca-pop-lgp) [reply Heading/Heading.module.less:34]: I think maybe Button update wasn't merged at that time and it didn't render the correct color for the icon of this back button. I honestly can't remember exactly. I will remove it.

File: Heading/Heading.module.less (hunk 7 above)
Comment (stanca-pop-lgp) [reply Heading/Heading.module.less:11]: I will remove it. Thanks!

**Review by MikyungKim — APPROVED** [review]
:lgtm:

**Pull request merged**